server/app.ts

```diff
@@ -0,0 +1,195 @@
+      rolling: true,
+    })
+  )
+
+  app.use(passport.initialize())
+  app.use(passport.session())
+
+  // Request Processing Configuration
+  app.use(bodyParser.json())
+  app.use(bodyParser.urlencoded({ extended: true }))
+
+  app.use(flash())
+
+  // Resource Delivery Configuration
+  app.use(compression())
+
+  // Cachebusting version string
+  if (production) {
+    // Version only changes on reboot
+    app.locals.version = version
+  } else {
+    // Version changes every request
+    app.use((req, res, next) => {
+      res.locals.version = Date.now().toString()
+      return next()
+    })
+  }
+
+  //  Static Resources Configuration
+  const cacheControl = { maxAge: config.staticResourceCacheDuration * 1000 }
+  ;[
+    '/assets',
+    '/assets/stylesheets',
+    '/assets/js',
+    '/node_modules/govuk-frontend/govuk/assets',
+    '/node_modules/govuk-frontend',
+    '/node_modules/jquery/dist',
+  ].forEach(dir => {
+    app.use('/assets', express.static(path.join(process.cwd(), dir), cacheControl))
+  })
+  ;['/node_modules/govuk_frontend_toolkit/images'].forEach(dir => {
+    app.use('/assets/images/icons', express.static(path.join(process.cwd(), dir), cacheControl))
+  })
+  ;['/node_modules/jquery/dist/jquery.min.js'].forEach(dir => {
+    app.use('/assets/js/jquery.min.js', express.static(path.join(process.cwd(), dir), cacheControl))
+  })
+
+  // Express Routing Configuration
+  app.get('/health', (req, res, next) => {
+    healthcheck(result => {
+      if (!result.healthy) {
+        res.status(503)
+      }
+      res.json(result)
+    })
+  })
+
+  // GovUK Template Configuration
+  app.locals.asset_path = '/assets/'
+
+  app.use((req, res, next) => {
+    res.locals.user = req.user
+    next()
+  })
+
+  // Don't cache dynamic resources
+  app.use(noCache())
+
+  // CSRF protection
+  if (!testMode) {
+    app.use(csurf())
+  }
+
+  // Update a value in the cookie so that the set-cookie will be sent.
+  // Only changes every minute so that it's not sent with every request.
+  app.use((req, res, next) => {
+    req.session.nowInMinutes = Math.floor(Date.now() / 60e3)
+    next()
+  })
+
+  app.get('/autherror', (req, res) => {
+    res.status(401)
+    return res.render('autherror')
+  })
+
+  app.get('/login', passport.authenticate('oauth2'))
+
+  app.get('/login/callback', (req, res, next) =>
+    passport.authenticate('oauth2', {
+      successReturnToOrRedirect: req.session.returnTo || '/',
+      failureRedirect: '/autherror',
+    })(req, res, next)
+  )
+
+  const authLogoutUrl = `${config.apis.hmppsAuth.externalUrl}/logout?client_id=${config.apis.hmppsAuth.apiClientId}&redirect_uri=${config.domain}`
+
+  app.use('/logout', (req, res) => {
+    if (req.user) {
+      req.logout()
+      req.session.destroy(() => res.redirect(authLogoutUrl))
+      return
+    }
+    res.redirect(authLogoutUrl)
+  })
+
+  app.use(authorisationMiddleware())
+  app.use('/', indexRoutes(standardRouter(userService)))
```

Comment:
not a big fan of this standardRouter stuff.. don't really understand what it brings other than unnecessary indirection?